### ISSUE TITLE
about duration predictor

### ISSUE THREAD

**Issue by wizardk** [keonlee9420/Comprehensive-Transformer-TTS#9]
In "learn_alignment: True" mode, the input of duration predictor is "x.detach() + self.predictor_grad * (x - x.detach())".

1. Why do we need detach()?
2. Why do we need add self.predictor_grad * (x - x.detach()) since it always is zero?

**Comment by keonlee9420** [comment]
Hi @wizardk , I think you can get a help from #8 .

**Comment by wizardk** [comment]
Thanks.
